src/engine/Operation.cpp

```diff
@@ -126,12 +133,7 @@ shared_ptr<const ResultTable> Operation::getResult(bool isRoot,
       // change in the DEBUG builds.
       AD_EXPENSIVE_CHECK(
           result.checkDefinedness(getExternallyVisibleVariableColumns()));
-      if (_timeoutTimer->wlock()->hasTimedOut()) {
-        throw ad_utility::TimeoutException(
-            "Timeout in " + getDescriptor() +
-            ". This timeout was not caught inside the actual computation, "
-            "which indicates insufficient timeout functionality.");
-      }
+      checkAbortion([this]() { return "In " + getDescriptor(); });
```

Comment:
This isn't guaranteed to work because of this:
https://github.com/ad-freiburg/qlever/pull/1125#discussion_r1376639690

There are currently uncancellable expensive operations in the code